From 4f1e00df5ea1e51042146f6277bb1af02f249cc3 Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentfredric@gmail.com>
Date: Sun, 11 Jun 2017 10:31:03 +1200
Subject: [PATCH] Fix for 5.26 removal of '.' in @INC

Previous scripts assumed CWD was in @INC and that "PARI::822" was in
wherever CWD was.

This fixes both of those, and the latter, because its not obvious
from the build system where one should assume CWD to be.

Instead, the absolute path of the directory of these scripts is
inserted into @INC
---
 src/desc/gen_member | 4 ++++
 src/desc/gen_proto  | 4 ++++
 src/desc/merge_822  | 4 ++++
 3 files changed, 12 insertions(+)

diff --git a/src/desc/gen_member b/src/desc/gen_member
index 2f3cf52..66dea2b 100755
--- a/src/desc/gen_member
+++ b/src/desc/gen_member
@@ -1,4 +1,8 @@
 #!/usr/bin/perl -w
+use File::Spec;
+use File::Basename qw( dirname );
+use lib dirname(File::Spec->rel2abs(__FILE__));
+
 use PARI::822;
 
 $class='member_functions';
diff --git a/src/desc/gen_proto b/src/desc/gen_proto
index 1b507fa..2fb4d3f 100755
--- a/src/desc/gen_proto
+++ b/src/desc/gen_proto
@@ -1,4 +1,8 @@
 #!/usr/bin/perl -w
+use File::Spec;
+use File::Basename qw( dirname );
+use lib dirname(File::Spec->rel2abs(__FILE__));
+
 use PARI::822;
 
 $class=$ARGV[0];
diff --git a/src/desc/merge_822 b/src/desc/merge_822
index a02d612..9ca8252 100755
--- a/src/desc/merge_822
+++ b/src/desc/merge_822
@@ -1,4 +1,8 @@
 #!/usr/bin/perl -w
+use File::Spec;
+use File::Basename qw( dirname );
+use lib dirname(File::Spec->rel2abs(__FILE__));
+
 use PARI::822;
 
 PARI::822::read(\%funcs,$_,1)
-- 
2.13.1

